### DISCUSSION TITLE
Getting Password error

### DISCUSSION THREAD

**Discussion by m-vinod-kumar** [community/community#42912] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

Hi all 

i am getting like this i dont why how to fix it i need without password can any one help me 

Cloning into 'infrastructure-modules'...
git@mucsggit02.sg.de.pri.o2.com's password:

Permission denied, please try again.
git@mucsggit02.sg.de.pri.o2.com's password:

Permission denied, please try again.
git@mucsggit02.sg.de.pri.o2.com's password:
git@mucsggit02.sg.de.pri.o2.com: Permission denied (publickey,gssapi-keyex,gssapi-with-mic,password).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

**Comment by m-vinod-kumar** [comment]
i have access

**Comment by jfullstackdev** [comment]
you can access the private repo via GitHub, to check whether you really have an access to that, you can have screenshot too to be posted here, that way we are sure that it's not about access issue

**Comment by SusheelThapa** [comment]
*I didnot get [git@mucsggit02.sg.de.pri.o2.com](mailto:git@mucsggit02.sg.de.pri.o2.com) what do you mean by this*
*And   i think this is not the github repository clone link*


## Access Right

First check if you have correct access right to the repository or not

Having access right mean that, you have deploy you ssh key

To check if you have depoly or not refer [here](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/checking-for-existing-ssh-keys)

If the ssh key isn't deploy you can create one refering [here](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)

And finally add the created ssh key in github, refering [here](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)

## Cloning

- Then, if you repository is private, using ssh to while clone as 
   ```sh
    git clone git@github.com:<username>/<repo_name>.git
   ```
- If you repository is public, use http method 
   ```sh
    https://github.com/<username>/<repo_name>.git
    ```
